### PR TITLE
vSphere Cloud Provider Documentation

### PR DESCRIPTION
**Pull request by andrewsykim**
**What this PR does / why we need it**:
Iterating on the vSphere cloud provider docs as per https://github.com/kubernetes/cloud-provider/issues/9, https://github.com/kubernetes/cloud-provider-vsphere/issues/41 and https://github.com/kubernetes/cloud-provider-vsphere/issues/146. The PR is currently just a template/outline and has no content. Once we agree on a general outline we can start working on the content.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
https://github.com/kubernetes/cloud-provider/issues/9
https://github.com/kubernetes/cloud-provider-vsphere/issues/41

**Special notes for your reviewer**:
This doc covers both the in-tree and out-of-tree vSphere cloud provider and assumes the reader is relatively new to using Kubernetes. 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by andrewsykim** [comment]
@cantbewong @embano1 @frapposelli @dougm @dvonthenen @mylesagray

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  - [kube-controller-manager](#kube-controller-manager)
+  - [cloud-controller-manager](#cloud-controller-manager)
+  - [kubeadm](#kubeadm)
+- [Architecture](#architecture)
+  - [Kubernetes using the In-Tree vSphere Provider](#kubernetes-using-the-in-tree-vsphere-provider)
+  - [Kubernetes using the Out-of-Tree vSphere Provider](#kubernetes-using-the-out-of-tree-vsphere-provider)
+- [vSphere Integrations](#vsphere-integrations)
+  - [Kubernetes Nodes](#kubernetes-nodes)
+  - [Kubernetes Zones/Regions Topology](#kubernetes-zones-regions-topology)
+  - [Kubernetes LoadBalancers](#kubernetes-loadbalancers)
```
Comment (andrewsykim) [inline docs/README.md]: We don't support this yet but I think worth calling that out + future intentions

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  - [Kubnernetes Routes](#kubernetes-routes)
```
Comment (andrewsykim) [inline docs/README.md]: We don't support this yet but I think worth calling that out + future intentions

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- [Example Configurations and Manifests](#example-manifests)
+- [Tutorials](#tutorials)
+- [Addons](#addons)
+  - [Storage](#storage)
```
Comment (codenrhoden) [inline docs/README.md]: I like the callouts for the addons. Let's just make sure that when we do fill this with content, we make it clear that the out-of-tree provider *requires* the CSI plugin.

Comment (andrewsykim) [reply docs/README.md]: Good point, how about renaming this to `Required Plugins` and linking out to CSI docs?

Comment (codenrhoden) [reply docs/README.md]: Sounds good -- we may want both categories: optional and required?

Comment (andrewsykim) [reply docs/README.md]: How about `Required` and `Recommended`? I feel too many things fall into optional plugins 😆

**Comment by andrewsykim** [comment]
@mylesg filled out the first few sections, left VMware specific component definitions for you to fill out

**Comment by andrewsykim** [comment]
Oops, meant to ping @mylesagray sorry

**Pull request merged**